src/pal/src/exception/signal.cpp

```diff
@@ -67,56 +70,67 @@ typedef void *siginfo_t;
 #endif  /* !HAVE_SIGINFO_T */
 typedef void (*SIGFUNC)(int, siginfo_t *, void *);
 
+#if !HAVE_MACH_EXCEPTIONS
 // Return context and status for the signal_handler_worker.
 struct SignalHandlerWorkerReturnPoint
 {
     bool returnFromHandler;
     CONTEXT context;
 };
+#endif // !HAVE_MACH_EXCEPTIONS
 
 /* internal function declarations *********************************************/
 
+#if !HAVE_MACH_EXCEPTIONS
 static void sigill_handler(int code, siginfo_t *siginfo, void *context);
 static void sigfpe_handler(int code, siginfo_t *siginfo, void *context);
 static void sigsegv_handler(int code, siginfo_t *siginfo, void *context);
 static void sigtrap_handler(int code, siginfo_t *siginfo, void *context);
 static void sigbus_handler(int code, siginfo_t *siginfo, void *context);
 static void sigint_handler(int code, siginfo_t *siginfo, void *context);
 static void sigquit_handler(int code, siginfo_t *siginfo, void *context);
+#endif // !HAVE_MACH_EXCEPTIONS
 static void sigterm_handler(int code, siginfo_t *siginfo, void *context);
 
+#if !HAVE_MACH_EXCEPTIONS
 static bool common_signal_handler(int code, siginfo_t *siginfo, void *sigcontext, int numParams, ...);
 
 #ifdef INJECT_ACTIVATION_SIGNAL
 static void inject_activation_handler(int code, siginfo_t *siginfo, void *context);
 #endif
+#endif // !HAVE_MACH_EXCEPTIONS
 
 static void handle_signal(int signal_id, SIGFUNC sigfunc, struct sigaction *previousAction, int additionalFlags = 0);
 static void restore_signal(int signal_id, struct sigaction *previousAction);
 
 /* internal data declarations *********************************************/
 
+#if !HAVE_MACH_EXCEPTIONS
 struct sigaction g_previous_sigill;
 struct sigaction g_previous_sigtrap;
 struct sigaction g_previous_sigfpe;
 struct sigaction g_previous_sigbus;
 struct sigaction g_previous_sigsegv;
 struct sigaction g_previous_sigint;
 struct sigaction g_previous_sigquit;
+#endif // !HAVE_MACH_EXCEPTIONS
 struct sigaction g_previous_sigterm;
 
 static bool registered_sigterm_handler = false;
```

Comment:
A nit - similar to what I've said above, could you please move the two variables above all the `struct sigaction xxxx` declarations above and use single `#if !HAVE_MACH_EXCEPTIONS`?